protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
> What if the Server doesn't support partialResult

This should read _if true and the server supports streaming_

> What if a method doesn't want to use Partial Result

All methods should behave consistently and either all use `$/partialResult` or none. This makes the receiver code less complicated. This shouldn't make it more complicated for the sender because your protocol layer in the sender implementation should just be able to turn a full result of a method handler into a single replace op.